### PR TITLE
ENH: AT2K2 transmission calculations on PLC [LCLSPC-208]

### PR DESCRIPTION
**Pull request by klauer**
We have determined it's not acceptable for PMPS to have a Python process calculating transmission values as [solid-attenuator](https://github.com/pcdshub/solid-attenuator) is currently doing in the case of AT2K2. 

As such, using a new dependency [lcls-twincat-physics](https://github.com/pcdshub/lcls-twincat-physics), the solid attenuators must calculate transmission on the PLC to coordinate with PMPS at PLC rates.

This PR is a first-pass attempt at performing these calculations. 

A few things:
* I'd like some help in triple-checking that I have the filter information correct.
* The Python IOC allows the user to `caput` new materials and thicknesses as they please. This will need to be phased out, and the PLC will be the canonical, read-only source of that information.
* Python will still be in control of choosing the filters to insert for the requested transmission value
* We will need to coordinate with Maggie to get `rPhotonEnergy` sent along on the EtherCAT bus from the arbiter.
* We'll also need some time to test that this all works.

To consider/check:
* Some of the general DUTs/FBs provided here may be moved to lcls2-cc-lib
* Summary/total transmission (product of all transmissions)
* Transmission of 1.0 in "out" state

### CODE REVIEW

**Comment by klauer** [comment]
The calculation methods have been unit tested, but the combined filter material information + thickness + tying it to a specific state has not.

@jyotiphy, depending on your availability, I'd like to get this activated and tested in the near term. A bit of coordination with @ghalym is also required because the KFE arbiter needs modifications to send along `rPhotonEnergy` via EtherCAT as well.

I can't guarantee it'll go well in the first attempt, so follow-up PRs and additional testing may be required.

**Comment by jyotiphy** [comment]
@klauer This looks good to me. I haven't revive the rix motion PLC and IOC host yet but planning to do that later in the week. But I think first we need to have changes deployed on KFE arbiter to read 'rPhotonEnergy' ? I know we do that on rix optics PLC, but not sure exactly how and if we need cable going from arbiter to PLC ? Let me know what works for you and Maggie.

**Comment by klauer** [comment]
Thanks, @jyotiphy 👍  Later this week is just fine for me, at least.

I'm not sure if the arbiter needs updating _before_ this IOC, but @ghalym may be able to clarify.

**Comment by ghalym** [comment]
@klauer The arbiter needs to be updated the same time you are activating config on the motion plc., just to confirm together that everything is working fine. Later this week works for me too.
@jyotiphy no extra cables are necessary, assuming that the rix motion is already connected to the arbiter via EL6695.

**Comment by jyotiphy** [comment]
I forgot about this task and also was away Thu-Fri last week. RIX motion PLC is up and even IOC host is up as I was testing some motion. 
@ghalym Yes, I double-checked there's a cable connection motion PLC and arbiter via EL6695
Let me know when you would like to deploy and test this code.

**Comment by klauer** [comment]
I'd be OK with some time this afternoon, if that works for you two.

**Comment by ghalym** [comment]
Today @4PM?

**Comment by jyotiphy** [comment]
Yes, this should work. I have a meeting in the afternoon, should be quick, but please go ahead if I can't make it. I will check and leave the PLC running on master brach.

**Comment by klauer** [comment]
Sounds good. 4 pm it is, then. (And sorry random GitHub user '4pm' for dragging you in here ;) )

**Comment by ghalym** [comment]
oops!!

**Comment by klauer** [comment]
OK, I think this is good for now on the PLC side; there will be additional functionality for this in the future, though. 

TODO:
* IOC needs rebuilding

File: plc-kfe-rix-motion/kfe_rix_motion/POUs/AT2K2/FB_SXR_SATT_Stage.TcPOU [hunk 1]
```diff
@@ -18,12 +18,43 @@ VAR_INPUT
...
 END_VAR
 VAR_OUTPUT
     // stExtra : DUT_MotionStage_Extras;
     fTemp1 : LREAL;
     fTemp2 : LREAL;
     bIsStationary : BOOL;
+    bError : BOOL;
+
+    {attribute 'pytmc' := '
+        pv: Material
```
Comment (ZLLentz) [inline plc-kfe-rix-motion/kfe_rix_motion/POUs/AT2K2/FB_SXR_SATT_Stage.TcPOU:34]: PV naming consistency thing, shouldn't these be e.g. `MATERIAL`?

**Review by ZLLentz — APPROVED** [review]
Confirmed that the materials/positions/thicknesses are consistent with the table at https://confluence.slac.stanford.edu/display/L2SI/AT2K2-SOLID+SXR+Solid+Attenuator

File: plc-kfe-rix-motion/kfe_rix_motion/POUs/AT2K2/FB_SXR_SATT_Stage.TcPOU (hunk 1 above)
Comment (klauer) [reply plc-kfe-rix-motion/kfe_rix_motion/POUs/AT2K2/FB_SXR_SATT_Stage.TcPOU:34]: I cycled between the two. Overall, I dislike `MATERIAL` but I think it's more consistent... Will change.

**Review by jyotiphy — APPROVED** [review]

**Pull request merged**